v2/brigadier/src/groups.ts

```diff
@@ -1,45 +1,93 @@
-import { Job } from "./jobs"
+import { Runnable } from "./runnables"
 
-export class Group {
+/**
+ * The base type for Runnables that contain multiple other Runnables.
+ * Do not construct the base Group type; use Job.sequential (or SerialGroup)
+ * or Job.concurrent (or ConcurrentGroup) instead.
+ * 
+ * @abstract
+ */
+class Group {
+  protected runnables: Runnable[]
 
-  public static async runAll(jobs: Job[]): Promise<void[]> {
-    const group = new Group(jobs)
-    return group.runAll()
+  protected constructor(...runnables: Runnable[]) {
+    this.runnables = runnables || []
   }
 
-  public static async runEach(jobs: Job[]): Promise<void> {
-    const group = new Group(jobs)
-    return group.runEach()
+  public add(...runnables: Runnable[]): void {
+    for (const runnable of runnables) {
+      this.runnables.push(runnable)
+    }
   }
 
-  private jobs: Job[]
+  public length(): number {
+    return this.runnables.length
+  }
+}
 
-  public constructor(jobs?: Job[]) {
-    this.jobs = jobs || []
+/**
+ * A Runnable that consists of sub-Runnables (such as jobs
```

Comment:
Non-blocking nit: "composed of" language might convey what this does more clearly.